src/main/java/com/spotify/autoscaler/ClusterStats.java

```diff
@@ -211,62 +202,70 @@ void setStats(BigtableCluster cluster, int currentNodes) {
 
     if (clusterData == null) {
       // First time we saw this cluster, register a gauge
-      this.registry.register(
-          APP_PREFIX
-              .tagged("what", "node-count")
-              .tagged("project-id", cluster.projectId())
-              .tagged("cluster-id", cluster.clusterId())
-              .tagged("instance-id", cluster.instanceId()),
-          (Gauge<Integer>)
-              () -> registeredClusters.get(cluster.clusterName()).getCurrentNodeCount());
-
-      this.registry.register(
-          APP_PREFIX
-              .tagged("what", "min-node-count")
-              .tagged("project-id", cluster.projectId())
-              .tagged("cluster-id", cluster.clusterId())
-              .tagged("instance-id", cluster.instanceId()),
-          (Gauge<Integer>) () -> registeredClusters.get(cluster.clusterName()).getMinNodeCount());
-
-      this.registry.register(
-          APP_PREFIX
-              .tagged("what", "max-node-count")
-              .tagged("project-id", cluster.projectId())
-              .tagged("cluster-id", cluster.clusterId())
-              .tagged("instance-id", cluster.instanceId()),
-          (Gauge<Integer>) () -> registeredClusters.get(cluster.clusterName()).getMaxNodeCount());
-
-      this.registry.register(
-          APP_PREFIX
-              .tagged("what", "effective-min-node-count")
-              .tagged("project-id", cluster.projectId())
-              .tagged("cluster-id", cluster.clusterId())
-              .tagged("instance-id", cluster.instanceId()),
-          (Gauge<Integer>)
-              () -> registeredClusters.get(cluster.clusterName()).getEffectiveMinNodeCount());
 
-      this.registry.register(
-          APP_PREFIX
-              .tagged("what", "last-check-time")
-              .tagged("project-id", cluster.projectId())
-              .tagged("cluster-id", cluster.clusterId())
-              .tagged("instance-id", cluster.instanceId()),
-          (Gauge<Long>)
-              () ->
-                  db.getBigtableCluster(
-                          cluster.projectId(), cluster.instanceId(), cluster.clusterId())
-                      .flatMap(
-                          p ->
-                              Optional.of(
-                                  Duration.between(
-                                      p.lastCheck().orElse(Instant.EPOCH), Instant.now())))
-                      .get()
-                      .getSeconds());
+      for (BigtableMetric.Metrics metric : BigtableMetric.Metrics.values()) {
+        Gauge metricValue = null;
+        switch (metric) {
+          case NODE_COUNT:
```

Comment:
maybe the enum `BigtableMetric.Metrics` can have a method that returns a lambda instead of this `switch` statement